Lib/javascript/v8/javascriptcode.swg

```diff
@@ -435,7 +478,12 @@ fail:
  * ----------------------------------------------------------------------------- */
 %fragment("jsv8_register_class", "templates")
 %{
+#if (V8_MAJOR_VERSION-0) < 4 && (SWIG_V8_VERSION < 0x031903) || (SWIG_V8_VERSION < 0x0706)
```

Comment:
This is very confusing because 2 different conventions for `SWIG_V8_VERSION` are used in the same expression. I'd use `V8_MAJOR_VERSION` and `V8_MINOR_VERSION` in the RHS, as the intention is to deprecate `SWIG_V8_VERSION` and use native v8 constants for node `>= 4.3.0`.

Reply:
For the future, I suggest to use `NODE_MODULE_VERSION`. See its [description](https://nodejs.org/en/download/releases/#ref-1)